turbopack/crates/turbopack-ecmascript/src/simple_tree_shake/mod.rs

```diff
@@ -0,0 +1,87 @@
+//! Intermediate tree shaking that uses global information but not good as the full tree shaking.
+
+use anyhow::{Context, Result};
+use rustc_hash::{FxHashMap, FxHashSet};
+use turbo_rcstr::RcStr;
+use turbo_tasks::{ResolvedVc, Vc};
+use turbopack_core::{module_graph::ModuleGraph, resolve::ExportUsage};
+
+use crate::chunk::EcmascriptChunkPlaceable;
+
+#[turbo_tasks::function]
+pub async fn get_module_export_usages(
+    graph: ResolvedVc<ModuleGraph>,
+    module: ResolvedVc<Box<dyn EcmascriptChunkPlaceable>>,
+) -> Result<Vc<ModuleExportUsageInfo>> {
+    let export_usage_info = compute_export_usage_info(graph)
+        .resolve_strongly_consistent()
+        .await?;
+
+    let export_usage_info = export_usage_info.await?;
+
+    let Some(exports) = export_usage_info.used_exports.get(&module) else {
+        // Pages like [project]/packages/next/dist/esm/build/templates/pages.js does not have any
+        // usage information.
+        return Ok(ModuleExportUsageInfo::all());
```

Comment:
Why not? Are these the roots of the module graph and thus don't have any incoming references?

Reply:
if a root has exports, why can't we still trim them?  We should just be preserving the side effects from entry points, right?